src/ucp/core/ucp_ep.c

```diff
@@ -76,7 +77,7 @@ ucs_status_t ucp_ep_add_pending_uct(ucp_ep_h ep, uct_ep_h uct_ep,
     }
 
     /* Forced progress */
-    status = req->func(req);
+    status = req->func(req, status);
```

Comment:
More generally, req->func is called with status = {UCS_OK or UCS_ERR_BUSY}. It looks like it should produce a status, but not accept one as input.